libutils/condition_macros.h

```diff
@@ -17,4 +26,12 @@
 }
 #endif
 
+// Similar to assert_or_return, except you put it inside the if
+// body which handles the error in release builds
+#ifndef debug_abort_if_reached
+#define debug_abort_if_reached() { \
+    assert(false);                 \
```

Comment:
I guess the only problem here is that you aren't told exactly what condition is wrong when it fails. Adding a parameter to the macro would kind of remove the point of having it in the first place, however.

Reply:
It should be placed inside an if, and you get a line number. It should be pretty clear.

Reply:
I hope you get a line number. Good point, when I statrt using it I can do some testing and improve it if necessary.